xtuner/entry_point.py

```diff
@@ -119,29 +115,104 @@
        for k, v in special.items()}
 }
 
+
+def list_cfg():
+    from xtuner.tools import list_cfg
+    return list_cfg.__file__
+
+
+def copy_cfg():
+    from xtuner.tools import copy_cfg
+    return copy_cfg.__file__
+
+
+def log_dataset():
+    from xtuner.tools import log_dataset
+    return log_dataset.__file__
+
+
+def check_custom_dataset():
+    from xtuner.tools import check_custom_dataset
+    return check_custom_dataset.__file__
+
+
+def train():
+    from xtuner.tools import train
+    return train.__file__
+
+
+def test():
+    from xtuner.tools import test
+    return test.__file__
+
+
+def chat():
+    from xtuner.tools import chat
+    return chat.__file__
+
+
+def mmbench():
+    from xtuner.tools import mmbench
+    return mmbench.__file__
+
+
+def pth_to_hf():
+    from xtuner.tools.model_converters import pth_to_hf
+    return pth_to_hf.__file__
+
+
+def merge():
+    from xtuner.tools.model_converters import merge
+    return merge.__file__
+
+
+def split():
+    from xtuner.tools.model_converters import split
+    return split.__file__
+
+
+def arxiv_preprocess():
+    from xtuner.tools.data_preprocess import arxiv as arxiv_preprocess
+    return arxiv_preprocess.__file__
+
+
+def convert_help_msg():
+    print_log(CONVERT_HELP_MSG, 'current')
+
+
+def preprocess_help_msg():
+    print_log(PREPROCESS_HELP_MSG, 'current')
+
+
 modes = {
-    'list-cfg': list_cfg.__file__,
-    'copy-cfg': copy_cfg.__file__,
-    'log-dataset': log_dataset.__file__,
-    'check-custom-dataset': check_custom_dataset.__file__,
-    'train': train.__file__,
-    'test': test.__file__,
-    'chat': chat.__file__,
-    'mmbench': mmbench.__file__,
+    'list-cfg': list_cfg,
+    'copy-cfg': copy_cfg,
+    'log-dataset': log_dataset,
+    'check-custom-dataset': check_custom_dataset,
+    'train': train,
+    'test': test,
+    'chat': chat,
+    'mmbench': mmbench,
     'convert': {
-        'pth_to_hf': pth_to_hf.__file__,
-        'merge': merge.__file__,
-        'split': split.__file__,
-        '--help': lambda: print_log(CONVERT_HELP_MSG, 'current'),
-        '-h': lambda: print_log(CONVERT_HELP_MSG, 'current')
+        'pth_to_hf': pth_to_hf,
+        'merge': merge,
+        'split': split,
+        '--help': convert_help_msg,
+        '-h': convert_help_msg
     },
     'preprocess': {
-        'arxiv': arxiv_preprocess.__file__,
-        '--help': lambda: print_log(PREPROCESS_HELP_MSG, 'current'),
-        '-h': lambda: print_log(PREPROCESS_HELP_MSG, 'current')
+        'arxiv': arxiv_preprocess,
+        '--help': preprocess_help_msg,
+        '-h': preprocess_help_msg
     }
 }
 
+HELP_FUNCS = [preprocess_help_msg, convert_help_msg]
+MAP_FILE_FUNCS = [
+    list_cfg, copy_cfg, log_dataset, check_custom_dataset, train, test, chat,
+    mmbench, pth_to_hf, merge, split, arxiv_preprocess
+]
```

Comment:
Why we need this MAP_FILE_FUNCS?

Reply:
This version is not used, it will be used in the next CLI optimization version.